### DISCUSSION TITLE
Dropped Frames + No video length markers

### DISCUSSION THREAD

**Discussion by NathanielH-snek** [raspberrypi/picamera2#680] [Q&A] [answered]
Hello all! I have two separate issues but I have decided to ask them both in this post. The first is that I'm dropping frames in my video recordings. I'm using a picam v3 noir in a lowlight environment to record video in a labratory environment. I've tried directly recording in the commandline using libcamera, using the high level video recording functions, low level video functions and even writing it into an mp4 and have always run into this issue. I'm not sure if this is a problem with the api or is a hardware issue but dropped frames ruin the temporal aspect that I need. The other problem which is less of an issue and more of a question is how to package? timestamp? the video streams so that they have a length. I can convert them later on with ffmpeg, but it'd be easier if i could do it in script and I couldn't seem to find documentation on it (possible missed something in the docs.) thanks everyone!

**Comment by davidplowman** [comment]
Hi, please can you give the `libcamera-vid` command that you have tried and which is dropping frames? Also, please can you say what kind of Pi you are using. Other information, such as the minimum resolution, framerate and bitrate that you need to achieve would also be helpful. Thanks.

**Reply by NathanielH-snek** [reply]
Thank you for your response. I am using a 4B with 4gb RAM
Im shooting for minimum 720p with 30 fps but 1080p 30fps would be ideal. I'm not really as concerned with getting anything nicer until I get it to work really. Bitrate can be anything clear enough to train and run through models via DeepLabCut
Here is the script
```python
from picamera2 import Picamera2, Preview
from picamera2.encoders import Quality
from picamera2.encoders import H264Encoder
from picamera2.outputs import FfmpegOutput
from time import sleep
from datetime import datetime

now = datetime.now()
timeMinutes = 180
timeSeconds = timeMinutes * 60
dateAndTime = now.strftime("%m,%d,%Y%H-%M-%S")
print("Enter Animal Name:")
animalName = str(input())
#filename = "/media/camerapi1/b125d8d6-317b-4b88-9dfd-216329074a5c/Data/Video" + "/" + animalName + ":" + dateAndTime + ".h264"
filename = "/media/camerapi1/camvideo/videodata" + "/" + animalName + "-" + dateAndTime + ".mp4"
print(filename)

picam2 = Picamera2()
video_config = picam2.create_video_configuration()
picam2.configure(video_config)

encoder = H264Encoder(10000000)
output = (filename)

picam2.start_preview(Preview.QTGL)
picam2.start_recording(encoder,output)
sleep(timeSeconds)
picam2.stop_recording()
picam2.stop_preview()
```

I have also tried using the quality settings High, Medium, and Very High
Here is the libcamera-vid command I've tried using
`libcamera-vid -t 7200000 -o libcamtest.h264`
I believe that is the function listing in the documentation.
Let me know if you want me to provide anything else

**Comment by davidplowman** [comment, marked as answer]
Hi, could you perhaps work through these [tips in the documentation](https://www.raspberrypi.com/documentation/computers/camera_software.html#high-framerate-capture) and report whether those help. Also, what camera are you using? Thanks!

**Reply by NathanielH-snek** [reply]
I overclocked the pi and that seems to have eliminated the dropped frames so far. I probably should have tried that earlier. I would however also like to fix the 0 length error. Is that simply a symptom of h264 files not having timestamp data? Is there a way that I can have this added when it is output as an mp4 file or is that something I should delve into ffmpeg for? Currently I remux every file with ffmpeg post hoc. Thanks.

**Reply by davidplowman** [reply]
That sounds likely. h264 files simple don't have proper timestamps, you can't tell how many frames is in them, they're simply not a proper "container" format.

You can create mp4 files directly with Picamera2, though in a number of respects libcamera-vid does this better because it has a tighter integration with the FFmpeg libraries.

**Reply by NathanielH-snek** [reply]
Ok, I will look into my options. Thank you for your help and pointing me in the right direciton
